code/modules/events/apc_short.dm

```diff
@@ -26,55 +26,108 @@
 /datum/event/apc_short/end()
 	return TRUE
 
-/proc/power_failure(announce=TRUE)
+/proc/depower_apcs()
 	var/list/skipped_areas_apc = list(
 		/area/engine/engineering,
 		/area/turret_protected/ai)
+	GLOB.event_announcement.Announce("Power failure detected in [station_name()]'s powernet. All APCs have lost power. Gravity systems likely to fail.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
+	for(var/thing in GLOB.apcs)
+		var/obj/machinery/power/apc/A = thing
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
+			continue
+		var/obj/item/stock_parts/cell/C = A.get_cell()
+		if(C)
+			C.charge = 0
+		current_area.power_change()
+	log_and_message_admins("Power has been drained from all APCs.")
 
+/proc/power_failure(announce=TRUE)
+	// skip any APCs that are too critical to disable
+	var/list/skipped_areas_apc = list(
+		/area/engine/engineering,
+		/area/turret_protected/ai)
 	if(announce)
 		GLOB.event_announcement.Announce("Overload detected in [station_name()]'s powernet. Engineering, please repair shorted APCs.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
-
 	// break APC_BREAK_PROBABILITY% of all of the APCs on the station
 	var/affected_apc_count = 0
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/C = thing
-		// skip any APCs that are too critical to disable
-		var/area/current_area = get_area(C)
-		if((current_area.type in skipped_areas_apc) || !is_station_level(C.z))
+	for(var/obj/machinery/power/apc/A in GLOB.apcs)
```

Comment:
You should leave this syntax how it was before:
```dm
for(var/thing in GLOB.apcs)
	var/obj/machinery/power/apc/A = thing
```
Ditto for other loops dealing with `GLOB.apcs` below you may have changed.

Reply:
Agreed! You pointed this out to me before and I thought I made the change back to the original syntax. I wonder if it reverted to my change (using a filtered for loop) when I closed and re-opened. I'll make the change again and see if it sticks. Ever since you explained this to me, I've been trying to make sure I code my loops in the way you have suggested.

Reply:
There's also `for(var/mob/living/M in GLOB.player_list)` on line 17. Is this redundant because GLOB.player_list is all living mob players? Could it just be `for(var/M in GLOB.player_list)`? or:

`for(var/M in GLOB.player_list)
var/mob/living/mob = M`

Reply:
Yeah so in this case it's necessary to have a filtered loop because the `GLOB.player_list` contains both alive and dead (ghost) mobs. However you *can* make it a typeless loop, but you'll need to check if `M` is not a dead mob first, then cast it to a living one. I'm not sure if that is actually any better though.